nexus-tweaks.user.js

```diff
@@ -32,6 +32,391 @@ function timesCharExist(x, c){
+          + '<div title="' + person.hpTooltip
+          + '" class="char-div-element hp-div hp-div-max" style="width: '
+          + (person.maxHp * scalingFactor) + 'px;"></div>'
+          + '<div title="' + person.hpTooltip
+          + '" class="char-div-element hp-div hp-div-current ' + hpClass
+          + '" style="width: ' + (person.currentHp * scalingFactor) + 'px;"></div>'
+          + '<div class="char-div-element hp-div hp-div-previous" '
+          + 'style="width: ' + (previousHp * scalingFactor) + 'px;"></div>'
+          + '<div class="status-images">' + statusTags + '</div>';
+
+      // optionally add MP bars if MP values known
+      if (person.maxMp > 0) {
+        resultHtml = resultHtml
+            + '<div title="' + person.hpTooltip
+            + '" class="char-div-element mp-div mp-div-max" style="width: '
+            + (person.maxMp * scalingFactor) + 'px;"></div>'
+            + '<div title="' + person.hpTooltip
+            + '" class="char-div-element mp-div mp-div-current" style="width: '
+            + (person.currentMp * scalingFactor) + 'px;"></div>';
+      }
+
+      // add segmentation for health bar (10hp = 1 segment)
+      for(let hpseg = 10;
+        hpseg < (Math.max(person.maxHp, person.currentHp) - 1); hpseg = hpseg + 10) {
+        resultHtml = resultHtml
+            + '<div class="char-div-element bar-segments-div" style="left: '
+            + (188 + hpseg * scalingFactor) + 'px;"></div>'
+      }
+      resultHtml = resultHtml
+          + '</div>';
+    }
+
+    var title = 'There ' + peopleMatch[1] + ' ' + personList.length
+      + ' other ' + peopleMatch[3] + ' here.';
+    const newPersonSpanCreator = (p =>
+      '<span class="politics-' + p.politics + '">'
+         + '<a href="https://nexusclash.com/clash.php?op=character&id=' + p.id
+         + '" target="_blank">' + p.name + '</a></span>');
+    if (peopleJoined.length > 0) {
+      title += ' New: ';
+      var joinedHtml = peopleJoined.map(newPersonSpanCreator);
+      title += joinedHtml.join(', ');
+    }
+
+    charArea.innerHTML = title + '<br><br>' + resultHtml + '<br>';
+    // necessary hack because of new float child elements in charAreaDiv
+    // this should also keep the petListArea div unchanged and intact
+    charArea.style.float = 'left';
+    charArea.style.width = '100%';
+
+    // remove all previously saved people that are no longer present
+    // so they can register as new again when they reappear
+    for(const personToDelete of peopleToDelete) {
+      localStorage.removeItem(personToDelete);
+    }
+  }
+
+  await mod.registerSetting(
+      'select',
+      'character-list-display-style',
+      'Display Style',
+      'Choose the character list tweak to use',
+      [{'value' : 'character-list-thalanor', 'text' : 'Thalanor\'s Graphical'},
+      {'value' : 'character-list-nexus-tweaks', 'text' : 'Default Nexus Tweaks'}]
+  )
+
+  await mod.registerMethod(
+      'async',
```

Comment:
comment: I would recommend making this method "sync" instead of async.
(That will involve removing the async syntax at the method definition too).

The difference is that sync methods run first, one at a time, in order.
Since this does some major DOM manipulation, you can avoid potential concurrency issues by taking advantage of this.

Reply:
made sync; please check if this is now correctly done (kept await at mod.registerMethod, removed async at characterList definition, used 'sync' as arg0 in registerMethod